File: 프로그래머스/2/17687. ［3차］ n진수 게임/［3차］ n진수 게임.py
def solution(n, t, m, p):
    answer = ''
    result = ""
    text_list = []
    hex_dict = {k:v for k,v in zip(range(16),"0123456789ABCDEF")}
    
    # 숫자 말하는 순서 : (p + (m * 현재턴(0~t-1)))
    # 숫자 진수로 변환 -> m * (t-1) 까지
    
    max_num = m * t
    rev_base = ''
    
    for i in range(max_num):
        num = i
        rev_base = ''
        
        if num == 0:
            rev_base += "0"
        while i > 0:
            i, mod = divmod(i, n)
            
            rev_base += hex_dict[mod]

        answer += "".join(rev_base[::-1])

    for j in range(t):
        result += answer[p + (m * j) - 1]
    
    
    return result

File: 프로그래머스/2/17687. ［3차］ n진수 게임/［3차］ n진수 게임.java
class Solution {
    
    private static final char[] DIGITS = 
    {'0','1','2','3','4','5','6','7','8','9','A','B','C','D','E','F'};
    
    public String solution(int n, int t, int m, int p) {
        int roop = t * m;
        StringBuilder sb = new StringBuilder();
        
        for (int i = 0; i <= roop; i++) sb.append(convertNum(i, n));
        
        StringBuilder answer = new StringBuilder();
        for (int i = p - 1; i < roop; i += m) answer.append(sb.charAt(i));
        
        return answer.toString();
    }
    
    private String convertNum(int num, int n) {
        if (num == 0) return "0";
        
        StringBuilder sb = new StringBuilder();
        
        while (num > 0) {
            sb.append(DIGITS[num % n]);
            num /= n;
        }
        
        return sb.reverse().toString();
    }
}